nucypher/network/nodes.py

```diff
@@ -281,7 +278,9 @@ def __init__(self,
 
         self._abort_on_learning_error = abort_on_learning_error
 
-        self.__known_nodes = self.tracker_class(domain=domain, this_node=self if include_self_in_the_state else None)
+        self.__known_nodes = self.tracker_class(
+            domain=self.domain, this_node=self if include_self_in_the_state else None
```

Comment:
Where is `self.domain` set? If not set above, this may also have effects on Porter which is a `Learner`.

Reply:
Great question == right now it's set in `Character.__init__`. I'm trying to be strategic about where to do the str -> domain lookup, perhaps porter can handle this on it's own?

Reply:
Ignoring Porter, logically it feels like the `domain` is appropriately a part of `Learner`...i.e. the existential question is, *what is a learner without a domain*?

Reply:
✔️